### ISSUE TITLE
Switch statements

### ISSUE THREAD

**Issue by kaidesu** [ghost-language/ghost#57]

**Comment by skx** [comment]
This might provide some inspiration:

* https://github.com/skx/monkey/pull/71

Of course specifics might vary, but I was quite pleased with how easily it fell out.  Although the regular-expression matching was a bit of a hack.

**Comment by kaidesu** [comment]
oh! thank you! I'll definitely check this out~

**Comment by kaidesu** [comment]
Two years later but I finally got around to the first pass of switch statements. Implemented in c6ec58a74b08089fd7457c8321da441cb934a2c5, bab9a1065052e687df7ff67528fb8ca35e500b73, 404f8d93b70ede02b62cf2d53482d22643fc7bcf, 7befed6bf1fe57ef9500749bc0fb6fe809ed4b48, and a32bdf1c9280513407f86c6017738c9beff75319.

Will continue to work on things - need to add default branches still. Thanks again for the inspiration @skx - it certainly did fall into place quite easily!
